src/javaObject.cpp

```diff
@@ -54,6 +67,18 @@
       v8::Handle<v8::String> methodNameSync = NanNew<v8::String>((methodNameStr + "Sync").c_str());
       v8::Local<v8::FunctionTemplate> methodCallSyncTemplate = NanNew<v8::FunctionTemplate>(methodCallSync, methodName);
       funcTemplate->PrototypeTemplate()->Set(methodNameSync, methodCallSyncTemplate->GetFunction());
+
+      if (promisifying) {
+        v8::Local<v8::Object> recv = asyncOptions; // just a dummy receiver
+        v8::Local<v8::Value> argv[] = { methodCallTemplate->GetFunction() };
+        v8::Local<v8::Value> result = promisify->Call(recv, 1, argv);
+        if (!result->IsFunction()) {
+          printf("Promisified result is not a function.\n");
```

Comment:
Yes, this check was partly just my debug diagnostics. But if a user passes in an invalid promisify function, one that doesn't return a function, then we should catch the error here. I agree that assert(false) in acceptable in this case.